### ISSUE TITLE
Agent incorrectly assumes that JVM version == JDK version

### ISSUE THREAD

**Issue by asm0dey** [DataDog/java-profiler#136]
We at BellSoft have JDK 8 with underlying JVM 17. If I run my Java8 application on my JDK with DataDog agent, I get a crash like [this one](https://github.com/user-attachments/files/17211655/he_err_pid1.log)

In reality, it's possible (and correct) to check the JVM version, not JDK. For example, the property `java.vm.version` contains `17.0.9+10-LTS` in our case.

It seems to be related to these lines: https://github.com/DataDog/java-profiler/blob/11f0ba72528ff254b1089f41c3be17f687963136/ddprof-lib/src/main/cpp/vmStructs.cpp#L592

**Comment by jbachorik** [comment]
Aargh ... 
Would it possible to find something distinctive? Java 8 will put `25.*` into that system property. While we could hard-code '25 == 8', once we get proper JDK 25, this will get incredibly messy.

I just want to make sure what would be the right cause of action - currently, we are reading `java.vm.version` to retrieve the JVM version. But it does return value 17, since you are running JVM 17 there. But the version returned by `java.runtime.version` would be '8.*' I assume, correct? And the code that is crashing is actually using the JDK functionality and as such we would have to have two versions - vm and jdk, right? And then check all occasions where we guard some functionality based on the Java version and decide whether it is JVM or JDK related.

**Comment by jbachorik** [comment]
> If it will help I can privately send you a build of our JDK for you to check everything you need

yes, this would be very helpful

**Comment by dpochepk** [comment]
> Aargh ... Would it possible to find something distinctive? Java 8 will put `25.*` into that system property. While we could hard-code '25 == 8', once we get proper JDK 25, this will get incredibly messy.
> 
> I just want to make sure what would be the right cause of action - currently, we are reading `java.vm.version` to retrieve the JVM version. But it does return value 17, since you are running JVM 17 there. But the version returned by `java.runtime.version` would be '8.*' I assume, correct? And the code that is crashing is actually using the JDK functionality and as such we would have to have two versions - vm and jdk, right? And then check all occasions where we guard some functionality based on the Java version and decide whether it is JVM or JDK related.

Yes. You're correct. java.runtime.version still has something like "1.8.0_*".

**Comment by asm0dey** [comment]
@jbachorik I've sent a mail to your address I found here on GH, could you please answer when you have a minute?

**Comment by jbachorik** [comment]
Fixed in #142
